clang/include/clang/Tooling/DependencyScanning/DependencyScanningFilesystem.h

```diff
@@ -168,6 +170,12 @@ class DependencyScanningFilesystemSharedCache {
     /// The backing storage for cached contents.
     llvm::SpecificBumpPtrAllocator<CachedFileContents> ContentsStorage;
 
+    /// Map from filenames to cached real paths.
+    llvm::StringMap<const CachedRealPath *> RealPathsByFilename;
```

Comment:
I guess this could be inlined into something like
```c++
llvm::StringMap<std::pair<const CachedFileSystemEntry *, const CachedRealPath *>> InfoByFilename
```

Any preferences?

Reply:
I would lean towards combining them unless there's a drawback I'm not seeing - the keys are fairly large here so it's nice to share storage for that.  I would maybe feel differently for a DenseMap of small keys :shrug: